docs/cloud_build_standards.md

```diff
@@ -0,0 +1,90 @@
+## Cloud build standards
+
+### Account Layout
+
+- Separate accounts for separate concerns i.e. CI, Ops, Prod etc. This is because it:
+  - decouples usage types into it's own account structure making it a more logical separation due to IAM being global
+  - prevents tag names possibly overlapping for common services.
+  - allows the audit of that account to be more identiable i.e. changes to prod were expected in the account due to known changes / releases etc.
+- Use cross account federation where possible for users
+- Have separate audit accounts with restricted policies on that accounts
+
+### Account naming
+
+The name of the AWS account needs to contain useful information, below is a list of things that should be contained within it
+
+| Type | Description | Usage |
+| ---- | ------ | ------ |
+| Directorate | Directorate Cost number | 1234 |
+| Cost Code | Cost code number | 3456789 |
+| Portfolio | Portfolio abbreviation | x |
+| Project Name | Name of the project | Projectx |
+| Env Name | Name of environment for account | test, notprod, prod, ops |
+
+The overall Format would be `DIRECTORATE-COSTCODE_PORTFOLIO-PROJECT-ENV`
+
+### IP allocation
+
+If the account is going to share other services i.e. any internal central service, then the IP allocation will need to be addressed. This is to avoid IP conflicts with other services, that may also be VPN / Peered to the central internal service.
+
+Please discuss this with the team when the account is created.
+
+### General Conventions
+
+If we follow the above, then most of the environment separation will be happening at an account level rather than at a resource level. Where there are multi-tenancy type services i.e. an account with multiple service environments for applications, then the env prefix will be necessary.
+
+Please familiarise yourself with
+
+- Tagging Concepts and what services support / don't support [tagging](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html)
+- [s3 naming restrictions](https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html)
+
+### Tagging considerations
+
+- Tags can be applied anytime
+- Cost allocation reports are only available from the point tags were activated
+- After creating a tag [key] for costs, it must be marked activated ? added as a cost allocation tag, otherwise it will not be visible in the Cost Explorer
+
+### General Tagging Standards
+
+Below are a list of tagging standards to consider. Not all will be applicable i.e. based on the above shared account concepts (it's hosted in an account with other services), if it's clustered or not.
+
+All of the below should apply however, to all resources where you can.
+
+
+| Type | Description | Usage | Optional |
+| ---- | ------ | ------ | ------|
+| `TYPE` | Service type | CI | No |
+| `ENV` | Type-ENV  | ci-dev, ops-prod, notprod-uat, notprod-ppd | No |
```

Comment:
The `ENV` here probably shouldn't include the Type, as that's separately recorded above. So:
TYPE: ops
ENV: prod

Reply:
cool i'll change, i thought the same but in some places we had it like that already